.gitlab-ci.yml

```diff
@@ -0,0 +1,73 @@
+
+# /************************************************************************
+ #  File: .gitlab-ci.yml
+ #  Author: mdr0id
+ #  Date: 9/10/2018
+ #  Description:  Used to setup runners/jobs for librustzcash
+ #  Usage: Commit source and the pipeline will trigger the according jobs.
+ #         For now the build and test are done in the same jobs.
+ #
+ #  Known bugs/missing features:
+ #
+ # ************************************************************************/
+  
+stages:
+  - build
+  - test
+  - deploy
+
+rust-latest:
+  stage: build
+  image: rust:latest
+  script:
+    - cargo --verbose --version
+    - time cargo build --verbose
+
+rust-nightly:
+  stage: build
+  image: rustlang/rust:nightly
+  script:
+    - cargo --verbose --version
+    - cargo build --verbose
+  allow_failure: true
+
+librustzcash-test-latest:
+  stage: test
+  image: rust:latest
+  script:
+    - cargo --verbose --version
+    - time cargo test --release --verbose
```

Comment:
If I understand correctly, this is going to have the following result:

- Build stage: Cargo version gets printed
- Build stage: Everything is built in debug mode
- Test stage: Cargo version gets printed again (why would it have changed in between?)
- Test stage: Everything is built in release mode and then tested.

I'm not sure we get much benefit from building both in debug and release mode. Release mode is slower to build, will run the tests (probably much) faster, and provide less information during test failures. Are there likely situations where we could end up being able to compile in release mode but not in debug mode (or vice versa) @ebfull?

Reply:
@str4d More or less, yes:
- Build stage against Rust latest, print cargo version, time and build librustzcash in debug mode(default)
- Build stage against Rust nightly, print cargo version, build librustzcash
- Test stage against Rust latest, print cargo version (it shouldn't change but wanted it there for completeness), build latest and test
- Test stage against Rust nightly, "  "

The `librustzcash-rust-rc` and `librustzcash-rust-nightly-rc` are if you wanted to manually deploy an artifact of librustzcash.

Only reason I put the granularity in the build was because IIRC there was a time George has an issue that only manifested itself between the debug/release mode.